monitor/extractors/windows.go

```diff
@@ -0,0 +1,144 @@
+// +build windows
+
+package extractors
+
+import (
+	"debug/pe"
+	"encoding/hex"
+	"fmt"
+	"os/user"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/shirou/gopsutil/process"
+	"go.aporeto.io/trireme-lib/common"
+	"go.aporeto.io/trireme-lib/policy"
+	"go.aporeto.io/trireme-lib/utils/cgnetcls"
+	portspec "go.aporeto.io/trireme-lib/utils/portspec"
+)
+
+// WindowsMetadataExtractorType is a type of Windows metadata extractors
+type WindowsMetadataExtractorType func(event *common.EventInfo) (*policy.PURuntime, error)
+
+// WindowsServiceEventMetadataExtractor is a windows service based metadata extractor
+func WindowsServiceEventMetadataExtractor(event *common.EventInfo) (*policy.PURuntime, error) {
+
+	runtimeTags := policy.NewTagStore()
+
+	for _, tag := range event.Tags {
+		parts := strings.SplitN(tag, "=", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid tag: %s", tag)
+		}
+		runtimeTags.AppendKeyValue("@usr:"+parts[0], parts[1])
+	}
+
+	userdata := WinProcessInfo(event.PID)
+
+	for _, u := range userdata {
+		runtimeTags.AppendKeyValue("@sys:"+u, "true")
+		runtimeTags.AppendKeyValue("@app:windows:"+u, "true")
+	}
+
+	runtimeTags.AppendKeyValue("@sys:hostname", findFQDN(time.Second))
+	runtimeTags.AppendKeyValue("@os:hostname", findFQDN(time.Second))
+
+	if fileMd5, err := ComputeFileMd5(event.Executable); err == nil {
+		runtimeTags.AppendKeyValue("@sys:filechecksum", hex.EncodeToString(fileMd5))
+		runtimeTags.AppendKeyValue("@app:windows:filechecksum", hex.EncodeToString(fileMd5))
+	}
+
+	depends := getDllImports(event.Name)
+	for _, lib := range depends {
+		runtimeTags.AppendKeyValue("@sys:lib:"+lib, "true")
+		runtimeTags.AppendKeyValue("@app:windows:lib:"+lib, "true")
+	}
+
+	options := policy.OptionsType{}
+	for index, s := range event.Services {
+		if s.Port != 0 && s.Ports == nil {
+			if pspec, err := portspec.NewPortSpec(s.Port, s.Port, nil); err == nil {
+				event.Services[index].Ports = pspec
+				event.Services[index].Port = 0
+			} else {
+				return nil, fmt.Errorf("Invalid Port Spec %s", err)
+			}
+		}
```

Comment:
we prefer the style 

if pspec, err := ... ; err != nil {
return 
}

continue with the non-error case.

Reply:
pspec needs to be used in the non-error case and so can't be defined local to the if block. but I can restructure to check err and return early.